src/apps/command_view_app.tsx

```diff
@@ -75,11 +74,20 @@ class CommandViewApp extends Component<MyProps, MyState> {
     this.uischema = build.form;
     this.initialData = build.model;
     let requestData = null;
-    if (this.props.location.state) {
+    let pourItAgainRequest: any = window.localStorage.getItem(
+      `lastKnown_pour_it_again_request`
+    );
+    pourItAgainRequest = JSON.parse(pourItAgainRequest);
+    if (pourItAgainRequest) {
       requestData = this.formatRequestToData(
-        this.props.location.state.request,
+        pourItAgainRequest.request,
         this.initialData
       );
+      console.log(pourItAgainRequest);
+      window.localStorage.setItem(
+        `lastKnown_pour_it_again_request`,
+        JSON.stringify(false)
```

Comment:
As a separate thought on the clearing of `lastKnown_pour_it_again_request`, there's a race condition created here where you could technically clear the value before the page you spawned goes to read it. Not (I don't think) in the simple case of a single page being spawned, but try quickly middle clicking the button twice and you'll notice that the second page that loads doesn't get the request data.

This specific behavior might not be a huge deal (you can, after all, just slow down your middle clicking), but I worry that it points to this general solution not being entirely safe.  You are, after all, essentially stashing the request info into a single global key / value pair for the site. Since someone could conceivably have any number of windows open, it's easy to imagine bad interactions happening as that single stashed value gets manipulated by different open windows.

My lack of experience with javascript development unfortunately means I can't tell you what the right solution is here, only that my gut tells me that this isn't it.